simplecpp.cpp

```diff
@@ -3219,8 +3219,9 @@ void simplecpp::preprocess(simplecpp::TokenList &output, const simplecpp::TokenL
             macros.insert(std::make_pair("__STDC_VERSION__", Macro("__STDC_VERSION__", std_def, files)));
         } else {
             std_def = simplecpp::getCppStdString(dui.std);
-            if (!std_def.empty())
-                macros.insert(std::make_pair("__cplusplus", Macro("__cplusplus", std_def, files)));
+            if (std_def.empty())
+                throw std::runtime_error("unknown standard specified: '" + dui.std + "'");
```

Comment:
I have the feeling that we do not want that this exception is unhandled by simplecpp::preprocess. It would be better to write a error in the output list and return.

Reply:
I believe we still want that simplecpp is designed to be portable and easy to reuse.

my gut feeling is that a library should not throw exceptions. if you use compiler A to compile the library and compiler B to compile the executable that uses the library.. is the exceptions "binary" interface well defined enough so A and B will always use the same?

Reply:
I had this sitting in my tree for ages now and I am close to have no pending local changes (finally) so I pushed this. There's no pressing reason for having this. It's just to be verify the given value is actually something valid.

I will take a look if we can handle this differently. Maybe needs some cleanups and definitely some more tests on the Cppcheck side first. The standard handling is questionable in some parts - there's several TODOs in place (or in my local tree).